src/components/modals/AddressModal.tsx

```diff
@@ -177,13 +181,93 @@ export class AddressModalComponent extends React.Component<Props, State> {
     return address
   }
 
+  // This approach considers resolution for tokens
+  getRnsChainIdFromPluginId = (pluginId: string): number => {
+    switch (pluginId) {
+      case 'rsk':
+        return 137
+      case 'ethereum':
+        return 60
+      case 'ethereumclassic':
+        return 61
+      case 'bitcoin':
+        return 0
+      case 'litecoin':
+        return 2
+      case 'dogecoin':
+        return 3
+      case 'dash':
+        return 5
+      case 'ripple':
+        return 144
+      case 'bitcoincash':
+        return 145
+      case 'binance':
+        return 714
+      case 'stellar':
+        return 148
+      case 'eos':
+        return 194
+      default:
+        return -1
+    }
+  }
+
+  fetchRnsAddress = async (domain: string): Promise<string> => {
+    try {
+      const pluginId = this.getRnsChainIdFromPluginId(this.props.coreWallet.currencyInfo.pluginId)
```

Comment:
this should be `chainId`

Reply:
@paullinator Thank you for the review. Changes made, let me know if something else needs improvement